aws-java-sdk-dynamodb/src/test/java/com/amazonaws/services/dynamodbv2/document/utils/ValueMapAndWithJsonSupportTest.java

```diff
@@ -0,0 +1,54 @@
+package com.amazonaws.services.dynamodbv2.document.utils;
+
+import static java.util.Arrays.asList;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.amazonaws.AmazonClientException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.hamcrest.CoreMatchers;
+import org.junit.Test;
+
+public class ValueMapAndWithJsonSupportTest {
+
+    private static final String NO_JSON_STRING = "nojson";
+    private static final String KEY = "somekey";
+
+    @Test(expected = AmazonClientException.class)
+    public void valueMapCreationShouldFailIfNoJsonStringIsUsedAsValue() {
+        new ValueMap().withJSON("a", NO_JSON_STRING);
+    }
+
+    @Test
+    public void valueMapShouldReturnAProperDeserializedJsonMap()  {
+        String json = "{ \"fruit\" : \"pear\" , \"color\" : \"green\" }";
+        Map<String, Object> expected = new LinkedHashMap<String, Object>();
+        expected.put("fruit", "pear");
+        expected.put("color", "green");
+
+        ValueMap valueMap = new ValueMap().withJSON(KEY, json);
+        Map<String,Object> actual = (Map<String,Object>)valueMap.get(KEY);
+        assertThat(actual, is(expected));
+    }
+
+    @Test
+    public void valueMapShouldReturnAProperDeserializedJsonList()  {
+        String json = "[\"red\",\"green\",\"blue\"]";
+        List<String>  expected = new ArrayList<String>(asList("red","green","blue"));
+
+        ValueMap valueMap = new ValueMap().withJSON(KEY, json);
+        List<String> actual = (List<String>)valueMap.get(KEY);
+        assertThat(actual.size(), is(expected.size()));
+        for (int counter = 0; counter < actual.size(); counter++) {
```

Comment:
Similarly we can get rid of the loop and the `expected` object and do something like:
```
assertThat(actual.size(), is(3));
assertThat(actual, hasItems("red","green","blue"));
```